astropy/wcs/wcsapi/wrappers/compound_wcs.py

```diff
@@ -0,0 +1,124 @@
+from functools import reduce
+
+import numpy as np
+
+from astropy.wcs.wcsapi import BaseLowLevelWCS, wcs_info_str
+
+__all__ = ['CompoundLowLevelWCS']
+
+
+def tuplesum(lists):
+    return reduce(tuple.__add__, map(tuple, lists))
+
+
+class CompoundLowLevelWCS(BaseLowLevelWCS):
+    """
+    A wrapper that takes multiple low level WCS objects and makes a compound
+    WCS that combines them.
+
+    Parameters
+    ----------
+    *wcs : `~astropy.wcs.wcsapi.BaseLowLevelWCS`
+        The WCSes to combine
```

Comment:
This probably could do with a little more detail on ordering? I assume it's really pixel axes that we are in strict order here? Although obviously the world axes are not crossing the wcs lines at this point.